cli/azd/pkg/input/console.go

```diff
@@ -63,14 +67,35 @@ func (c *AskerConsole) SetWriter(writer io.Writer) {
 
 // Prints out a message to the underlying console write
 func (c *AskerConsole) Message(ctx context.Context, message string) {
-	// Only write to the console during interactive & non-formatted responses.
-	if c.interactive && (c.formatter == nil || c.formatter.Kind() == output.NoneFormat) {
+	if c.formatter != nil && c.formatter.Kind() == output.JsonFormat {
+		// we do not expect formatter.Format to fail on valid input. Discard the error to appease the linter (note
+		// that we implicitly discard the error from fmt.Fprintln later, as well).
+		_ = c.formatter.Format(c.consoleMessageForMessage(message), c.writer, nil)
+	} else if c.interactive && (c.formatter == nil || c.formatter.Kind() == output.NoneFormat) {
 		fmt.Fprintln(c.writer, message)
 	} else {
 		log.Println(message)
 	}
 }
 
+// jsonObjectForMessage creates a json object representing a message. Any ANSI control sequences from the message are
+// removed. A trailing newline is added to the message.
+func (c *AskerConsole) consoleMessageForMessage(message string) contracts.ConsoleMessage {
+	// Strip any ANSI colors for the message.
+	var buf bytes.Buffer
+
+	// We do not expect the io.Copy to fail since none of these sub-calls will ever return an error (other than
+	// EOF when we hit the end of the string)
+	if _, err := io.Copy(colorable.NewNonColorable(&buf), strings.NewReader(message)); err != nil {
+		panic(fmt.Sprintf("jsonObjectForMessage: did not expect error from io.Copy but got: %v", err))
+	}
+
+	// Add the newline that would have been added by fmt.Println when we wrote the message directly to the console.
+	buf.WriteByte('\n')
```

Comment:
I wonder if this is what we should be doing.. If we think about the output being machine-readable, I think formatting should be handled by the UI layer. We'd may consider instead to ensure that no formatting characters are present.

Reply:
I thought a lot about this, given that I'm already stripping the color codes.  I think the problem is if we remove these sort of control characters, the simple algorithm of "just take the text and write it to 'the console'" falls over, because now you have to figure out where to insert newlines and the like.

This approach allows us to at least shuttle the console output we would have written to consumers so they can display it as they please.

The other option would be for us to say there should not be embedded control characters like '\n' in the strings we pass to `.Message` and perhaps introduce a `.Messageln` so we can get the formatting right.  Do you think that would be better?